### DISCUSSION TITLE
Switch ESLint for Deno Lint

### DISCUSSION THREAD

**Discussion by OmerWow** [vercel/next.js#38286] [Ideas]
### Describe the feature you'd like to request

Hi,

Since Next.js is very performance oriented and investing in Rust for its toolchain with SWC as the compiler and minifier, I think it'll be cool if you can replace ESlint with Deno Lint.

Deno uses it in production since I think forever and looks like it has many of ESlint's rules, the only problem (maybe) is that I'm not sure how easy it to code custom rules for it.

Anyway, it looks production ready, actively maintained and is over **112x times faster** than ESlint, taking about 0.02 seconds instead of 2.25 seconds in the example provided by the deno team.
Also, it's written in Rust!

Link: https://github.com/denoland/deno_lint

### Describe the solution you'd like

If migration is relatively easy, I think replacing ESLint with Deno Lint will be great.

I know it's possible for Deno Lint to work with other environments like in Node.js, and is **not** limited to Deno.

### Describe alternatives you've considered

One alternative is obviously ESlint which you're already using, which works great but is pretty slow.

Another alternative is RSlint, which is extremely fast and written in Rust, but is not production ready.
Overall Deno Lint seems more stable, more maintained and even faster based on some benchmarks I read.

**Comment by SukkaW** [comment]
Deno lint sounds great, but it might break all existing users with their own custom ESLint config. Also, does Deno lint support JSX and `eslint-config-react` yet?

**Comment by OmerWow** [comment]
> README.md

Here's the README.md link: https://github.com/denoland/deno_lint

**Comment by OmerWow** [comment]
> Deno lint sounds great, but it might break all existing users with their own custom ESLint config. Also, does Deno lint support JSX and `eslint-config-react` yet?

This is from the README.md: 
```
Supports recommended set of rules from ESLint and @typescript-eslint out of the box with no config.
```

Also, the Deno team just released a full-stack framework called Fresh (https://fresh.deno.dev/), it uses Deno as the backend and Preact (which is very similar to React and uses JSX) for the frontend, and they use Deno Lint as the linter for this framework.

So looks like it supports JSX and React pretty good from first look.

Regarding breaking custom configs, I guess you're right, but it may be pretty easy to make the conversion.
Also, I guess that in any case, with such a big change, it should be marked as 'experimental' and opt-in only via a next.config.js setting at the beginning.

**Comment by dandv** [comment]
I've also really wanted to like `deno lint` but:
1. It doesn't support any stylistic rules, as it considers them part of formatting (and `deno fmt` isn't part of the LSP)
2. It [misses some important rules, like `curly`](https://github.com/denoland/deno/discussions/27493), which can detect logical errors.
